### PR TITLE
Rustup "Remove `BodyAndCache`"

### PR DESCRIPTION
**Pull request by lzutao**
cc https://github.com/rust-lang/rust/pull/71044
changelog: none

### CODE REVIEW

File: clippy_lints/src/redundant_clone.rs [hunk 1]
```diff
@@ -146,7 +145,7 @@ impl<'a, 'tcx> LateLintPass<'a, 'tcx> for RedundantClone {
                 // `arg` is a reference as it is `.deref()`ed in the previous block.
                 // Look into the predecessor block and find out the source of deref.
 
-                let ps = mir_read_only.predecessors_for(bb);
+                let ps = mir.predecessors_for(bb);
```
Comment (ecstatic-morse) [inline clippy_lints/src/redundant_clone.rs:148]: FYI, `predecessors_for` is going away soon as well. The replacement would be:

```suggestion
                let ps = mir.predecessors()[bb];
```

**Comment by ecstatic-morse** [comment]
Thanks @lzutao

File: clippy_lints/src/redundant_clone.rs (hunk 1 above)
Comment (ecstatic-morse) [reply clippy_lints/src/redundant_clone.rs]: Whoops. Forgot a borrow.

Comment (lzutao) [reply clippy_lints/src/redundant_clone.rs]: I added the `&`.

**Pull request merged**